jenkins-builds/tools/mf.sh

```diff
@@ -0,0 +1,176 @@
+    mfn=`echo $mf |cut -d/ -f1`
+    echo $mf >> $tmp0
+    listofdeps0=`grep " load " $outf0 |awk -Fload '{print $2}' |tr -d " "`
+    if [ -n "$listofdeps0" ] ;then
+
+        for deps0 in $listofdeps0 ;do 
+            echo "\"$mf\" -> \"$deps0\";"
+
+            # --- sublevel1
+            module show $deps0 > $outf1 2>&1
+            grep -q "ERROR:105" $outf1 ;rc=$?
+            listofdeps1=`grep " load " $outf1 |awk -Fload '{print $2}' |tr -d " "`
+            if [ -n "$listofdeps1" ] ;then
+                for deps1 in $listofdeps1 ;do
+                    echo "\"$deps0\" -> \"$deps1\";"
+
+                    # --- sublevel2
+                    module show $deps1 > $outf2 2>&1
+                    grep -q "ERROR:105" $outf2 ;rc=$?
+                    listofdeps2=`grep " load " $outf2 |awk -Fload '{print $2}' |tr -d " "`
+                    if [ -n "$listofdeps2" ] ;then
+                        for deps2 in $listofdeps2 ;do
+                            echo "\"$deps1\" -> \"$deps2\";"
+                        done
+                    fi # --- sublevel2
+
+                done
+            fi # --- sublevel1
+
+        done
+    fi # --- sublevel0
+
+done |sort -u >> $0.dot 
+
+echo '}' >> $0.dot
+
+echo "module load stat"
+echo 'export LD_LIBRARY_PATH=/opt/cray/pe/stat/default/lib:$LD_LIBRARY_PATH'
+echo "dot -Tpng $0.dot > $0.png"
+echo "circo -Teps ./$0.dot > $0.eps"
+echo "try tulip too"
+#echo "dot -Tjpg $0.dot > $0.jpg"
+echo '------------------------------------------------------------------'
+
+#set -x
+# iterate through the list of software (=level0)
+for l0 in `sort -u $tmp0` ;do
+
+#echo continue;read
+    #echo "Listing l0=$l0 ..."
+    unset listl1a
+    unset listl1b
+    unset listl2a
+    unset listl2b
+    # iterate through the list of deps (=level1)
+    listofl1deps=`grep "^\"$l0\" ->" mf.sh.dot |awk -F"->" '{print $2}' |tr -d \;`
+    for l1 in $listofl1deps ;do 
+        # is this the end ?
+        rc=`grep -m1 -q "^$l1 ->" mf.sh.dot;echo $?`
+        if [ $rc -ne 0 ] ;then
+            # no more dep = end of the chain
+            listl1a="${listl1a} $l1 "
+        else
+            # more dep down the chain = continue parsing
+            l1tmp=`grep -m1 "^$l1 ->" mf.sh.dot |awk -F"->" '{printf "%s ",$1}' |tr -d \;`
+            listl1b=${listl1b}$l1tmp
+        fi        
+    done       
+    listl1s=`echo $listl1a $listl1b |tr " " "\n" |sort -u |awk '{printf "%s ",$0}'`
+    #echo "## l0=$l0: $listl1s"
+    echo '------------------------------------------------------------------'
+
+    # iterate through the list of deps'deps (=level2)
+    unset listl2s listl2 l2tmp 
+    for l1 in $listl1b ;do        
+        l2tmp=`grep "^$l1 ->" mf.sh.dot |awk -F"->" '{print $2}' |tr -d \;`
+        listl2=${listl2}$l2tmp
+        listl2s=`echo $listl2 |tr " " "\n" |sort -u |awk '{printf "%s ",$0}'`
+#ok        listofl2deps=`grep "^$l1 ->" mf.sh.dot |awk -F"->" '{print $2}' |tr -d \;`
```

Comment:
How about explaining why you keep all these lines commented out?

Reply:
d370e2b18c1ffcd88fd76137d4177da9e4dafeff